doc/1.4/language.md

```diff
@@ -3447,6 +3475,57 @@ in each (gated_write, write_1_clears) { is gated_write_1_clears; }
 
 The following is an example of the second case:
 ```
+template gated_write is write {
+    method write_allowed() -> (bool) default {
+        return true;
+    }
+
+    method write(uint64 val) default {
+        if (write_allowed()) {
+            base_write_of_gated_write(val); // instead of calling default()
+        }
+    }
+
+    method base_write_of_gated_write(uint64 val) default {
+        // If no conflicting template is in play such as to override this
+        // base method, then its behavior is as though the write implementation
+        // called default() instead
+        this.templates.write.write(val);
+    }
+}
+
+template write_1_clears is (write, get) {
+    method write(uint64 val) default {
+        default(get() & ~val);
+    }
+}
+
+template gated_write_1_clears is (gated_write, write_1_clears) {
+    method write(uint64 val) default {
+        // This makes gated_write the first link in the call chain...
+        this.templates.gated_write.write(val);
+    }
+
+    // And by overriding gated_write's base method, we make write_1_clears
+    // the second (and final) link in the chain
+    //
+    // If even more conflicting templates could be in play, then one could
+    // consider converting write_1_clears to also leverage a base method
+    // instead of calling default and extend the call chain; on the other
+    // hand, the way write_1_clears manipulates the written value may violate
+    // the expectations of implementations later on in the chain, so it might
+    // make most sense to always place write_1_clears as the final link, and
+    // have all other templates leverage base methods instead.
```

Comment:
I think the code snippet should explain itself only, no what-ifs. If this remark is needed, then we can place it after the snippet instead.

Reply:
The point is to touch upon the "thought is needed as to ordering" remark with a concrete example as to how it may matter.